molten-cache/src/test/java/com/hotels/molten/cache/ReactiveCacheContract.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hotels.molten.cache;
+
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import lombok.Value;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+/**
+ * {@code ReactiveCache} contract to test the expected behaviours.
+ */
+public abstract class ReactiveCacheContract {
```

Comment:
I would rather make this test not abstract and add a simple fake `ReactiveCache` implementation to make this work. What do you think?

Reply:
Or adding all the implementations in a sort of parameterized test based on JUnit 5.

Reply:
I'm a big fan of JUnit 5 but molten uses TestNG if I'm not mistaken.
The contract test idea is coming from actually Junit5: https://junit.org/junit5/docs/current/user-guide/#writing-tests-test-interfaces-and-default-methods
The reason I created an abstract class and not a specific test class is that each `ReactiveCache` implementation must satisfy this contract. The default methods introduced in this PR can be overridden, so it is reasonable to execute exactly the same test code against each implementation. With this approach extending this contract by all the implementations' test class would be the solution as I described in another comment.
The same approach could be used to test `get` and `post` methods.

Reply:
Molten supports JUnit 5 tests either, though earlier tests are truly written on TestNG.

`ReactiveCacheContract` as **an interface** with default methods looks good to me. 👍 Please migrate at least one (ex. the `ReactiveMapCacheTest`) to JUnit 5 as well to be able to add the contract, and I'll do it for the rest of the cache tests in a separate PR. 

Is that okay for you?

Reply:
Okay. Committed the change.

Reply:
@adambriny Can you please check it?

Reply:
Sure...